rllib/core/models/configs.py

```diff
@@ -0,0 +1,201 @@
+        if framework == "torch":
+            from ray.rllib.core.models.torch.mlp import TorchMLPModel
+
+            return TorchMLPModel(self)
+        else:
+            from ray.rllib.core.models.tf.mlp import TfMLPModel
+
+            # Activation functions in TF are lower case
+            self.output_activation = self.output_activation.lower()
+            self.hidden_layer_activation = self.hidden_layer_activation.lower()
+
+            return TfMLPModel(self)
+
+
+@ExperimentalAPI
+@dataclass
+class MLPEncoderConfig(MLPModelConfig):
+    """Configuration for an MLP that acts as an encoder.
+
+    See ModelConfig for usage details.
+    """
+
+    @_framework_implemented()
+    def build(self, framework: str = "torch") -> Encoder:
+        if framework == "torch":
+            from ray.rllib.core.models.torch.encoder import TorchMLPEncoder
+
+            return TorchMLPEncoder(self)
+        else:
+            from ray.rllib.core.models.tf.encoder import TfMLPEncoder
+
+            # Activation functions in TF are lower case
+            self.output_activation = self.output_activation.lower()
+            self.hidden_layer_activation = self.hidden_layer_activation.lower()
+
+            return TfMLPEncoder(self)
+
+
+@ExperimentalAPI
+@dataclass
+class LSTMEncoderConfig(ModelConfig):
+    """Configuration for a LSTM encoder.
+
+    See ModelConfig for usage details.
+
+    Attributes:
+        input_dim: The input dimension of the network. It cannot be None.
+        hidden_dim: The size of the hidden layer.
+        num_layers: The number of LSTM layers.
+        batch_first: Wether the input is batch first or not.
+        output_activation: The activation function to use for the output layer.
+        observation_space: The observation space of the environment.
+        action_space: The action space of the environment.
+        view_requirements_dict: The view requirements to use if anything else than
+        observation_space or action_space is to be encoded. This signifies an
+        anvanced use case.
+        get_tokenizer_config: A callable that returns a ModelConfig to build
+        tokenizers for observations, actions and other spaces that might be present
+        in the view_requirements_dict.
+
+    """
+
+    input_dim: int = None
+    hidden_dim: int = None
+    num_layers: int = None
+    batch_first: bool = True
+    output_activation: str = "linear"
+    observation_space: gym.Space = None
+    action_space: gym.Space = None
+    view_requirements_dict: ViewRequirementsDict = None
+    get_tokenizer_config: Callable[[gym.Space, Dict], ModelConfig] = None
+
+    @_framework_implemented(tf2=False)
+    def build(self, framework: str = "torch") -> Encoder:
+        if framework == "torch":
+            from ray.rllib.core.models.torch.encoder import TorchLSTMEncoder
+
+            return TorchLSTMEncoder(self)
+
+
+@ExperimentalAPI
+@dataclass
+class IdentityConfig(ModelConfig):
```

Comment:
Sorry, just renamed this to "IdentityEncoderConfig".
This can be, in theory, used if people just want completely distinct actor and critic networks.
That is, networks that have a different architecture.
I'll add this to a docstring.

Reply:
Are we gonna use this in RLllib code base? If it's just left for examples, let's remove it in this PR? I think shared = True / False already does that, don't it?